### DISCUSSION TITLE
(Re-)import aliases for batch update of description and recipient

### DISCUSSION THREAD

**Discussion by xanoni** [anonaddy/anonaddy#378] [Feature Requests]
I noticed that aliases can be exported as CSV in the settings.

Is there a way to delete all current aliases and re-import them from the same CSV format?

This would make it much easier to fix things like the format of the alias descriptions.

I haven't seen an import function, so in absence thereof I guess the best I could do is to request all aliases via `GET api/v1/aliases` and then update each of them individually via `PATCH api/v1/aliases/{id}`?

I would still encourage implementation of an import function, given many users are probably not super technical.

**API:**
https://app.anonaddy.com/docs/#get-all-aliases
https://app.anonaddy.com/docs/#update-a-specific-alias

**Comment by xanoni** [comment]
So to be specific, what I'm trying to achieve is to at least update the "Description" fields via some regex / `sed` filter.

What would be nice is to then also update the forwarding addresses based on what's in the updated description.

For that, I assume I would use `POST api/v1/alias-recipients` (in the absence of the import function that I'm asking about above).

**API:**
https://app.anonaddy.com/docs/#update-recipients-for-a-specific-alias

**Comment by willbrowningme** [comment]
I will be adding the ability to bulk import custom domain aliases but I don't think I'll be adding it for the other alias types as there are limits depending on plan to the number of aliases that can be created per hour.

You are correct you can update the description of a specific alias by using this API endpoint - https://app.anonaddy.com/docs/#aliases-PATCHapi-v1-aliases--id-

I'll soon be adding the option to the web application to bulk update recipients for up to 25 different aliases at once. So this should make it easier and quicker to do.
